src/main/java/com/nha/abdm/wrapper/hip/hrp/discover/DiscoveryService.java

```diff
@@ -26,14 +27,21 @@
 public class DiscoveryService implements DiscoveryInterface {
 
   @Autowired PatientRepo patientRepo;
-  @Autowired RequestManager requestManager;
+  private final RequestManager requestManager;
+  private final SessionManager sessionManager;
   @Autowired RequestLogService requestLogService;
   ErrorResponse errorResponse = new ErrorResponse();
   JaroWinkler jaroWinkler = new JaroWinkler();
 
   @Value("${onDiscoverPath}")
   public String onDiscoverPath;
 
+  @Autowired
+  public DiscoveryService(RequestManager requestManager, SessionManager sessionManager) {
```

Comment:
Since we are injecting instances via constructor, injecting all the other dependencies (`patientRepo`, `requestLogService`) as well via constructor would be good. As this would help out in writing test classes to inject Mock instances easily.

Reply:
I can take that as next cleanup task

Reply:
Sure, can you please keep a track of it (so that we don't loose it), by including this comment in that issue.